pkgs/fc/agent/src/fc/util/nixos.py

```diff
@@ -324,6 +324,32 @@ def update_system_channel(channel_url, log=_log):
         raise ChannelUpdateFailed(stdout=stdout, stderr=stderr)
 
 
+def find_nix_eval_warnings(stderr: str) -> list[str]:
+    """Returns the evaluation warnings raised by either `builtins.warn` or the
+    NixOS module warning system `config.warnings`.
+    Evaluation warnings raised in e.g. package definitions are not accessible via
+    the `config.warnings` module endpoint, hence need to be grabbed from stderr.
+    """
+
+    WARNINGS_MARKER = "evaluation warning: "
+    # As long as lix lacks a `builtins.warn`, the `lib.warn` NixOS code falls
+    # back to traces
+    WARNINGS_MARKER_LEGACY = "trace: evaluation warning: "
```

Comment:
Do we actually have Lix deployed anywhere? (I think the compatibility is worth keeping, as `builtins.warn` is a relatively new addition to Nix in any case.)